lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -149,8 +151,19 @@ impl GelfDeserializer {
                     let vector_val: value::Value = val.into();
                     log.insert(path!(key.as_str()), vector_val);
                 } else {
-                    return Err(format!("'The value type for field {}' is an invalid type. Additional field values \
-                                       should be either strings or numbers.", key).into());
+                    let type_ = if val.is_null() {
+                        "null"
+                    } else if val.is_array() {
+                        "array"
+                    } else if val.is_object() {
+                        "object"
+                    } else if val.is_boolean() {
+                        "boolean"
+                    } else {
+                        "unknown"
+                    };
```

Comment:
Could this just be:

```
let type_ = match val {
  serde_json::value::Value::Null => "null',
  ...
}
```

? I seem to remember you mentioning we can't match on serde_json::value::Value before, but I'm not actually seeing why 🤔

Reply:
Ah, nvm, I see the variants are not public.

Reply:
Hmm it seems I was mistaken! Thank you for your determination 😄